partner_compassion/models/partner_compassion.py

```diff
@@ -183,6 +187,29 @@ def name_search(self, name, args=None, operator='ilike', limit=80):
         return self.browse(to_ret_ids).name_get()
 
     ##########################################################################
+    #                             ONCHANGE METHODS                           #
+    ##########################################################################
+
+    @api.onchange('lastname', 'birthdate_date', 'firstname')
+    def _onchange_partner(self):
+        db_info = self.search([('firstname', '=', self.firstname),
+                               ('lastname', '=', self.lastname),
+                               ('birthdate_date', '=', self.birthdate_date)])
+        list_hom = []
+        for inf in db_info:
+            list_hom.append((4, inf.id))
+        self.listehomonyme = list_hom
+        self.homonymeformat = list_hom
```

Comment:
You can change all the code of the method up to here by :
```
self.partner_duplicate_ids = self.search([
    '|', ('email', '=', self.email),
    '&', '&', ('firstname', 'ilike', self.firstname),
    ('lastname', 'ilike', self.lastname),
    ('zip', '=', self.zip)
])
```
The search filters is equivalent to saying "All partner with same e-mail OR (firstname and lastname like AND zip equal).
You can directly assign the result to your Many2many field without having to create a list of (4, id) tuples. The (4, id) tuple is only useful when calling the method `write` on a recordset. When you have a single record (which is the case in api.onchange methods), you can use assignation like `self.name = 'Test name'` or put recordset in relational fields.